lib/scheduler.js

```diff
@@ -177,31 +176,23 @@ class Schedule {
 
   // Picks up the job from the queue and determines whether it
   // should be run or not. See `shouldRunJob()` below.
-  _processQueuedJob(job) {
-    const time = job.time;
-
-    if (shouldRunJob(this, job))
-      return this._runJob(time);
-    else {
+  _runJob(job) {
```

Comment:
FWIW the code was better when these were two separate responsibilities, even better if they were labeled correctly.